packages/material-ui/src/Modal/Modal.js

```diff
@@ -112,7 +112,9 @@ const Modal = React.forwardRef(function Modal(props, ref) {
     }
   });
 
-  const handleRendered = useEventCallback(() => {
+  const handlePortalRef = useEventCallback(node => {
+    mountNodeRef.current = node;
+
     if (onRendered) {
       onRendered();
```

Comment:
Not sure what the expected semantics are. Render is easily confused with a react render which wouldn't make sense and is also not correct in the current implementation. We should follow through and remove that as well.

It's probably easier to just remove it without a deprecation. Using the ref has a different behavior than the current `onRendered` semantics. We should just warn against that usage because it probably doesn't behave as one expects it to behave.

Reply:
Oh yeah, +1 for removing onRendered in v5 from the Modal too. The ref and onRedered behavior should be identical with these changes.